livre/eq/render/RayCastRenderer.cpp

```diff
@@ -237,67 +241,107 @@ struct RayCastRenderer::Impl
+        positions.emplace_back( minPos[0], maxPos[1], minPos[2] );
+
+        positions.emplace_back( minPos[0], maxPos[1], minPos[2] );
+        positions.emplace_back( maxPos[0], maxPos[1], minPos[2] );
+        positions.emplace_back( maxPos[0], minPos[1], minPos[2] );
+
+        // FRONT
+        positions.emplace_back( maxPos[0], maxPos[1], maxPos[2] );
+        positions.emplace_back( minPos[0], maxPos[1], maxPos[2] );
+        positions.emplace_back( minPos[0], minPos[1], maxPos[2] );
+
+        positions.emplace_back( minPos[0], minPos[1], maxPos[2] );
+        positions.emplace_back( maxPos[0], minPos[1], maxPos[2] );
+        positions.emplace_back( maxPos[0], maxPos[1], maxPos[2] );
+
+        // LEFT
+        positions.emplace_back( minPos[0], maxPos[1], minPos[2] );
+        positions.emplace_back( minPos[0], minPos[1], minPos[2] );
+        positions.emplace_back( minPos[0], minPos[1], maxPos[2] );
+
+        positions.emplace_back( minPos[0], minPos[1], maxPos[2] );
+        positions.emplace_back( minPos[0], maxPos[1], maxPos[2] );
+        positions.emplace_back( minPos[0], maxPos[1], minPos[2] );
+
+        // RIGHT
+        positions.emplace_back( maxPos[0], maxPos[1], maxPos[2] );
+        positions.emplace_back( maxPos[0], minPos[1], maxPos[2] );
+        positions.emplace_back( maxPos[0], minPos[1], minPos[2] );
+
+        positions.emplace_back( maxPos[0], minPos[1], minPos[2] );
+        positions.emplace_back( maxPos[0], maxPos[1], minPos[2] );
+        positions.emplace_back( maxPos[0], maxPos[1], maxPos[2] );
+
+        // BOTTOM
+        positions.emplace_back( maxPos[0], minPos[1], maxPos[2] );
+        positions.emplace_back( minPos[0], minPos[1], maxPos[2] );
+        positions.emplace_back( minPos[0], minPos[1], minPos[2] );
+
+        positions.emplace_back( minPos[0], minPos[1], minPos[2] );
+        positions.emplace_back( maxPos[0], minPos[1], minPos[2] );
+        positions.emplace_back( maxPos[0], minPos[1], maxPos[2] );
+
+        // TOP
+        positions.emplace_back( maxPos[0], maxPos[1], minPos[2] );
+        positions.emplace_back( minPos[0], maxPos[1], minPos[2] );
+        positions.emplace_back( minPos[0], maxPos[1], maxPos[2] );
+
+        positions.emplace_back( minPos[0], maxPos[1], maxPos[2] );
+        positions.emplace_back( maxPos[0], maxPos[1], maxPos[2] );
+        positions.emplace_back( maxPos[0], maxPos[1], minPos[2] );
     }
 
+    void renderVBO( bool front, bool back )
+    {
+        if( !front && !back )
+            return;
+
+        else if( front && !back )
+            glCullFace( GL_BACK );
+
+        else if( !front && back )
+            glCullFace( GL_FRONT );
+
+        glBindBuffer( GL_ARRAY_BUFFER, _posVBO );
+        glEnableVertexAttribArray( 0 );
+        glVertexAttribPointer( 0, 3, GL_FLOAT, GL_FALSE, 0, NULL );
+        glDrawArrays( GL_TRIANGLES, _currentIndex * 36, 36 );
+        _currentIndex++;
```

Comment:
This variable looks bad. The code will only work under strong and undocumented assumptions. renderBrick() takes a NodeId as a second parameter, can't the proper index be derived from that and passed to renderVBO()?